### PR TITLE
Fix issues with  historical builds

### PR DESCRIPTION
**Pull request by jjohannes**

### CODE REVIEW

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/GradleExecuterBackedSession.groovy [hunk 1]
```diff
@@ -55,7 +55,7 @@ class GradleExecuterBackedSession implements GradleSession {
                 } else {
                     runner.run()
                 }
-            } catch (Exception e) {
+            } catch (Throwable e) {
```
Comment (jjohannes) [inline subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/GradleExecuterBackedSession.groovy]: Ok. It was a kind of better-save-than-sorry change. I changed it back

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/AbstractToolingApiCrossVersionPerformanceTest.groovy [hunk 2]
```diff
@@ -225,8 +226,14 @@ abstract class AbstractToolingApiCrossVersionPerformanceTest extends Specificati
...
                     println "Run #${n + 1}"
-                    def measuredOperation = timer.measure {
-                        toolingApi.withConnection(action)
+                    def measuredOperation
+                    try {
+                        measuredOperation = timer.measure {
+                            toolingApi.withConnection(action)
+                        }
+                    } catch (Throwable e) {
+                        measuredOperation = new MeasuredOperation()
```
Comment (jjohannes) [inline subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/AbstractToolingApiCrossVersionPerformanceTest.groovy]: Ah. Missed that. Thanks for pointing out. Reverted this change.

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/AbstractToolingApiCrossVersionPerformanceTest.groovy [hunk 3]
```diff
@@ -263,9 +270,15 @@ abstract class AbstractToolingApiCrossVersionPerformanceTest extends Specificati
                         experimentSpec.listener.beforeInvocation(info)
                     }
                     println "Warm-up #${n + 1}"
-                    toolingApi.withConnection(action)
+                    def operation = null
```
Comment (jjohannes) [inline subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/AbstractToolingApiCrossVersionPerformanceTest.groovy]: Changed to use ` timer.measure` here.

**Comment by oehme** [comment]
There are still `catch Throwable` blocks. Throwable will not be thrown by the executor, it only uses unchecked Exceptions to signify build failures afaik. Catching `Throwable` would mask real issues on our fixtures.

**Comment by jjohannes** [comment]
Sorry I thought you were only referring to the one specific `catch` earlier.

I made the general change to `Throwable` because missing classes may cause `java.lang.AssertionError: java.lang.NoClassDefFoundError:`:
See: https://builds.gradle.org/viewLog.html?buildId=2300451

**Comment by oehme** [comment]
For that specific case we should put a `Class.forName("org/gradle/tooling/BuildAction")` inside the specific test before it actually starts calling the TAPI. That way we turn it from an `Error` into a normal `Exception`.

**Comment by jjohannes** [comment]
There is a bit too much Groovy magic in that code for my taste. :)

I have removed the change to throwable and added this:
https://github.com/gradle/gradle/pull/1401/commits/0e8b1b0c289c2504799015baa8950f52b5013c92

I have started another test build to verify.

**Comment by oehme** [comment]
Yes, it's hard to wrap your head around that one. I think there is one layer of indirection too much in TAPI performance fixtures, but that's something we can fix later.

**Comment by jjohannes** [comment]
Ok this one worked: https://github.com/gradle/gradle/pull/1401/commits/ae18319836435173412c4d9fcadea4c2b1dffb64

Here is a green build: (note I ran that one only for `2.1`)
https://builds.gradle.org/viewLog.html?buildId=2310212

**Pull request merged**